pkg/development/assertions.go

```diff
@@ -34,6 +34,24 @@ func runAssertions(devContext *DevContext, assertions []blocks.Assertion, expect
 	// TODO(jschorr): Support caveats via some sort of `assertMaybe`?
 	for _, assertion := range assertions {
 		tpl := tuple.MustFromRelationship(assertion.Relationship)
+
+		tplString, err := tuple.String(tpl)
+		if err != nil {
+			return nil, err
+		}
+
+		if tpl.Caveat != nil {
+			failures = append(failures, &devinterface.DeveloperError{
+				Message: fmt.Sprintf("cannot specify a caveat on an assertion: `%s`", tplString),
```

Comment:
Probably a question for a different PR, but what if one wants to assert a `CONDITIONAL` response? in such case you may want to specify in the assertion the context keys that weere missing.

Reply:
That will be done in a followup under a different section, likely named `assertConditional`